hasktorch/src/Torch/Static.hs

```diff
@@ -36,7 +38,7 @@ import qualified Torch.DType as DType
 natValI :: forall n. KnownNat n => Int
 natValI = fromIntegral $ natVal $ Proxy @n
 
-class (All KnownNat shape) => KnownShape shape where
+class KnownShape shape where
```

Comment:
not needed, does nothing

Reply:
I may be wrong, but IIUC this is necessary, because Haskell will happily accept types like `(1 - 5)` (`-` is simply not defined for those arguments) and will only raise an error if you ask that type to additionally satisfy `KnownNat`

Reply:
oh, I wasn’t aware of that. I just saw the instance hierarchy for `KnownShape`, and didn’t see a reason for the `All KnownNat` bound here. Since the everything built just fine without this bound, maybe we should add a test case that probes for simple `Nat` arithmetic at the type level and add this bound back. I’ll gladly make a PR for that! thanks for bringing this up :+1: